### PR TITLE
chore: update datafusion, arrow, prost, tonic, pbjson, etc

### PR DESCRIPTION
**Pull request by alamb**
# Rationale:
1. Get arrow 7
2. Get tonic / prost grades

# Changes:
1. Update to https://github.com/apache/arrow-datafusion/commit/794b92b4e03ee3b470c0506019bca4659cee5e34
2. Update our pins to arrow / arrow-flight and parquet to 7.0.0 as well
3. Update tonic/prost and crates that depends on them (e.g. pbjson)
4. Update to parquet-format --> 4.0

### CODE REVIEW

**Review by alamb — COMMENTED** [review]
@tustvold  I can't update `pbjson-build` for two crates because when I do I get a compile error

When I apply this:
```diff
diff --git a/generated_types/Cargo.toml b/generated_types/Cargo.toml
index e1df57b3..e2d958bf 100644
--- a/generated_types/Cargo.toml
+++ b/generated_types/Cargo.toml
@@ -24,7 +24,7 @@ num_cpus = "1.13.0"
 [build-dependencies] # In alphabetical order
 tonic-build = "0.6"
 prost-build = "0.9"
-pbjson-build = "0.1"
+pbjson-build = "0.2"
 
 [features]
 default = []
diff --git a/parquet_catalog/Cargo.toml b/parquet_catalog/Cargo.toml
index f9dc5e94..05d0ed10 100644
--- a/parquet_catalog/Cargo.toml
+++ b/parquet_catalog/Cargo.toml
@@ -20,7 +20,7 @@ parquet = "7.0"
 parquet_file = { path = "../parquet_file" }
 parquet-format = "4.0"
 parking_lot = "0.11.1"
-pbjson-types = "0.1"
+pbjson-types = "0.2"
 persistence_windows = { path = "../persistence_windows" }
 predicate = { path = "../predicate" }
 prost = "0.9"
```

I get this error:

```

  --- stderr
  thread 'main' panicked at 'illegal variant - AggregateTypeNone', /Users/alamb/.cargo/registry/src/github.com-1ecc6299db9ec823/pbjson-build-0.2.1/src/resolver.rs:85:9
  stack backtrace:
     0: rust_begin_unwind
               at /rustc/f1edd0429582dd29cccacaf50fd134b05593bd9c/library/std/src/panicking.rs:517:5
     1: std::panicking::begin_panic_fmt
               at /rustc/f1edd0429582dd29cccacaf50fd134b05593bd9c/library/std/src/panicking.rs:460:5
     2: pbjson_build::resolver::Resolver::rust_variant
               at /Users/alamb/.cargo/registry/src/github.com-1ecc6299db9ec823/pbjson-build-0.2.1/src/resolver.rs:85:9
     3: pbjson_build::generator::enumeration::generate_enum::{{closure}}
               at /Users/alamb/.cargo/registry/src/github.com-1ecc6299db9ec823/pbjson-build-0.2.1/src/generator/enumeration.rs:29:32
     4: core::iter::adapters::map::map_fold::{{closure}}
               at /rustc/f1edd0429582dd29cccacaf50fd134b05593bd9c/library/core/src/iter/adapters/map.rs:84:28
     5: core::iter::traits::iterator::Iterator::fold
               at /rustc/f1edd0429582dd29cccacaf50fd134b05593bd9c/library/core/src/iter/traits/iterator.rs:2171:21
     6: <core::iter::adapters::map::Map<I,F> as core::iter::traits::iterator::Iterator>::fold
               at /rustc/f1edd0429582dd29cccacaf50fd134b05593bd9c/library/core/src/iter/adapters/map.rs:124:9
     7: core::iter::traits::iterator::Iterator::for_each
               at /rustc/f1edd0429582dd29cccacaf50fd134b05593bd9c/library/core/src/iter/traits/iterator.rs:737:9
     8: <alloc::vec::Vec<T,A> as alloc::vec::spec_extend::SpecExtend<T,I>>::spec_extend
               at /rustc/f1edd0429582dd29cccacaf50fd134b05593bd9c/library/alloc/src/vec/spec_extend.rs:40:17
     9: <alloc::vec::Vec<T> as alloc::vec::spec_from_iter_nested::SpecFromIterNested<T,I>>::from_iter
               at /rustc/f1edd0429582dd29cccacaf50fd134b05593bd9c/library/alloc/src/vec/spec_from_iter_nested.rs:56:9
    10: <alloc::vec::Vec<T> as alloc::vec::spec_from_iter::SpecFromIter<T,I>>::from_iter
               at /rustc/f1edd0429582dd29cccacaf50fd134b05593bd9c/library/alloc/src/vec/spec_from_iter.rs:33:9
    11: <alloc::vec::Vec<T> as core::iter::traits::collect::FromIterator<T>>::from_iter
               at /rustc/f1edd0429582dd29cccacaf50fd134b05593bd9c/library/alloc/src/vec/mod.rs:2517:9
    12: core::iter::traits::iterator::Iterator::collect
               at /rustc/f1edd0429582dd29cccacaf50fd134b05593bd9c/library/core/src/iter/traits/iterator.rs:1745:9
    13: pbjson_build::generator::enumeration::generate_enum
               at /Users/alamb/.cargo/registry/src/github.com-1ecc6299db9ec823/pbjson-build-0.2.1/src/generator/enumeration.rs:24:28
    14: pbjson_build::Builder::generate
               at /Users/alamb/.cargo/registry/src/github.com-1ecc6299db9ec823/pbjson-build-0.2.1/src/lib.rs:219:21
    15: pbjson_build::Builder::build
               at /Users/alamb/.cargo/registry/src/github.com-1ecc6299db9ec823/pbjson-build-0.2.1/src/lib.rs:179:23
    16: build_script_build::generate_grpc_types
               at ./build.rs:115:5
    17: build_script_build::main
               at ./build.rs:12:5
    18: core::ops::function::FnOnce::call_once
               at /rustc/f1edd0429582dd29cccacaf50fd134b05593bd9c/library/core/src/ops/function.rs:227:5
  note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.

Compilation exited abnormally with code 101 at Thu Jan 13 08:50:51
```

Which appears to  be the same as https://github.com/influxdata/pbjson/issues/15, fixed in https://github.com/influxdata/pbjson/pull/17 but perhaps never released in the `0.2` line?

**Comment by alamb** [comment]
@tustvold  released a new version of pbjson and when I picked it up in b5457cb9 👍

File: Cargo.lock [hunk 1]
```diff
@@ -1282,9 +1301,9 @@ dependencies = [
  "grpc-router-test-gen",
  "observability_deps",
  "paste",
- "prost",
- "prost-build",
- "prost-types",
+ "prost 0.9.0",
```
[the remark this answers is not in the document]
Comment (alamb) [reply Cargo.lock]: https://github.com/mkmik/heappy/pull/13

**Review by mkmik — APPROVED** [review]

**Comment by alamb** [comment]
There is another failure I need to sort out:

https://app.circleci.com/pipelines/github/influxdata/influxdb_iox/14121/workflows/28a0ef75-dd40-44c3-b8bc-a639b3295732/jobs/106191

```
thread 'end_to_end_cases::operations_cli::test_start_stop' panicked at 'expected JSON output: Error("data did not match any variant of untagged enum Content", line: 10, column: 15)', influxdb_iox/tests/end_to_end_cases/operations_cli.rs:83:6
```

It seems to be related to parsing the output of a command as JSON. But I can't reproduce it outside the test harness. Will come back to it in a bit.

File: influxdb_iox/tests/end_to_end_cases/operations_cli.rs [hunk 2]
```diff
@@ -67,23 +67,30 @@ async fn test_start_stop() {
...
+        .unwrap()
+        .arg("operation")
+        .arg("wait")
+        .arg(name.to_string())
+        .arg("--host")
+        .arg(addr)
+        .assert()
+        .success();
+
+    // Can't parse a canceled operation due to
```
Comment (alamb) [inline influxdb_iox/tests/end_to_end_cases/operations_cli.rs:80]: This seems to be a bug in pbjson:  https://github.com/influxdata/influxdb_iox/issues/3458 but it doesn't strike me as a critical issue, so I am working around it in this PR

**Pull request merged**